source/common/http/http2/codec_impl.cc

```diff
@@ -349,6 +354,36 @@ void ConnectionImpl::StreamImpl::encodeMetadata(const MetadataMapVector& metadat
   }
 }
 
+void ConnectionImpl::StreamImpl::processBufferedData(bool stream_being_destroyed) {
+  ENVOY_CONN_LOG(debug, "Stream {} processing buffered data.", parent_.connection_, stream_id_);
+
+  auto next_stage = stream_manager_.getNextStage();
+  stream_manager_.flush_all_data_ = stream_being_destroyed;
+  bool continue_processing = stream_being_destroyed || !buffersOverrun();
+  while (continue_processing && next_stage.has_value()) {
+    switch (next_stage.value()) {
+    case BackedUpStreamManager::Stage::Body:
```

Comment:
this feels like it was written for the headers/body/trailers case but I think we can maybe simplify if it's just body and data.  I'd be inclined to 
while (data && coninue) {
  do data
}
if (continue) {
do trailers
}
WDYT?